modules/issue_tracker/php/endpoints/attachment.class.inc

```diff
@@ -0,0 +1,259 @@
+        default:
+            return new \LORIS\Http\Response\JSON\MethodNotAllowed(
+                $this->allowedMethods()
+            );
+        }
+    }
+
+    /**
+     * Handle GET requests for attachment
+     * and used for downloading attachment.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handleGET(ServerRequestInterface $request) : ResponseInterface
+    {
+        // Parse GET query params.
+        $values = $request->getQueryParams();
+
+        $file_hash = $values['file_hash'];
+        $filename  = $values['filename'];
+        $mime_type = $values['mime_type'];
+
+        $config = \NDB_Config::singleton();
+        $attachment_data_dir = rtrim(
+            $config->getSetting('IssueTrackerDataPath'),
+            '/'
+        );
+
+        $fileToDownload = trim($attachment_data_dir) .
+            '/attachments/' .
+            $file_hash;
+
+        if (!is_readable($fileToDownload)) {
+            throw new \LorisException('Attachment NOT FOUND');
+        }
+
+        $name = rawurldecode($filename);
+
+        return (new \LORIS\Http\Response())
+            ->withHeader('Content-Type', $mime_type)
+            ->withHeader(
+                'Content-Disposition',
+                'attachment; filename=' . basename($name)
+            )
+            ->withStatus(200)
+            ->withBody(new \LORIS\Http\FileStream($fileToDownload));
+    }
+
+    /**
+     * Handle POST requests for attachment
+     * and used for uploading attachment file.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handlePOST(ServerRequestInterface $request) : ResponseInterface
+    {
+        // Parse POST request body.
+        $values     = $request->getQueryParams();
+        $user       =& \User::singleton();
+        $attachment = new \LORIS\issue_tracker\UploadHelper();
+        $response   = $attachment->setupUploading(
+            $user,
+            $_FILES,
+            $values
+        );
+        return new \LORIS\Http\Response\JsonResponse(
+            $response
+        );
+    }
+
+    /**
+     * Handle DELETE requests for attachment
+     * and marks attachment as deleted from issue_tracker.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handleDELETE(ServerRequestInterface $request)
+    : ResponseInterface
+    {
+        // Verify User can delete file.
+        if ($this->_deletePermission($request)) {
+            // Parse request body.
+            $values    = $request->getQueryParams();
+            $file_hash = $values['file_hash'];
+            $DB        = \Database::singleton();
```

Comment:
Factory

Reply:
@johnsaigle thanks, I don't really know what the difference is here.